src/main/java/bio/terra/service/filedata/SnapshotCompute.java

```diff
@@ -0,0 +1,157 @@
+package bio.terra.service.filedata;
+
+import bio.terra.service.filedata.exception.DirectoryMetadataComputeException;
+import bio.terra.service.filedata.exception.FileNotFoundException;
+import bio.terra.service.filedata.google.firestore.FireStoreDirectoryEntry;
+import bio.terra.service.filedata.google.firestore.FireStoreFile;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.codec.digest.PureJavaCrc32C;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class SnapshotCompute {
```

Comment:
Would it be worth adding some unit/ small connected tests for these individual methods?